test/Graphics/VertexArray.cpp

```diff
@@ -0,0 +1,94 @@
+#include <SFML/Graphics/VertexArray.hpp>
+#include "GraphicsUtil.hpp"
+
+#include <doctest.h>
+
+TEST_CASE("sf::VertexArray class - [graphics]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::VertexArray vertexArray;
+            CHECK(vertexArray.getVertexCount() == 0);
+            CHECK(vertexArray.getPrimitiveType() == sf::PrimitiveType::Points);
+            CHECK(vertexArray.getBounds() == sf::FloatRect({0, 0}, {0, 0}));
+        }
+
+        SUBCASE("Explicit constructor with default argument")
+        {
+            const sf::VertexArray vertexArray(sf::PrimitiveType::Lines);
+            CHECK(vertexArray.getVertexCount() == 0);
+            CHECK(vertexArray.getPrimitiveType() == sf::PrimitiveType::Lines);
+            CHECK(vertexArray.getBounds() == sf::FloatRect({0, 0}, {0, 0}));
+        }
+
+        SUBCASE("Explicit constructor")
+        {
+            const sf::VertexArray vertexArray(sf::PrimitiveType::Lines, 10);
+            CHECK(vertexArray.getVertexCount() == 10);
+            CHECK(vertexArray.getPrimitiveType() == sf::PrimitiveType::Lines);
+            CHECK(vertexArray.getBounds() == sf::FloatRect({0, 0}, {0, 0}));
+            for (std::size_t i = 0; i < vertexArray.getVertexCount(); ++i)
+            {
+                CHECK(vertexArray[i].position == sf::Vertex().position);
+                CHECK(vertexArray[i].color == sf::Vertex().color);
+                CHECK(vertexArray[i].texCoords == sf::Vertex().texCoords);
+            }
+        }
+    }
+
+    SUBCASE("Resize array")
+    {
+        sf::VertexArray vertexArray;
+        vertexArray.resize(42);
+        CHECK(vertexArray.getVertexCount() == 42);
+        for (std::size_t i = 0; i < vertexArray.getVertexCount(); ++i)
+        {
+            CHECK(vertexArray[i].position == sf::Vertex().position);
+            CHECK(vertexArray[i].color == sf::Vertex().color);
+            CHECK(vertexArray[i].texCoords == sf::Vertex().texCoords);
+        }
+        vertexArray.clear();
+        CHECK(vertexArray.getVertexCount() == 0);
```

Comment:
Since clearing is a separate operation, I'd move it into a separate test, otherwise one doesn't really know that it's tested when looking at the cases